### DISCUSSION TITLE
Tasmota on Shelly 1V3+DHT22

### DISCUSSION THREAD

**Discussion by Massimo65** [arendst/Tasmota#17117] [Q&A]
Good morning

Thank you for your support, I have a Shelly 1v3 with addon connected to a DHT22 sensor. Can you describe me the correct configuration.

Thank you

**Comment by barbudor** [comment]
How is your DHT22 connected ? On which GPIO
Once you know that you can configure the appropriate GPIO in Configure -> Configure module to be of type AM2301 (compatible with DHT21 DHT22 AM2301 AM2302 AM2321)

**Comment by Massimo65** [comment]
Good morning,

I connected the DS18B20 via shelly addon and everything works fine, I replaced the DS18B20 with the DHT22 still connected via shelly addon but no configuration seems to work.
I then connected the DHT22 directly to the shelly 1v3 debug port on GPIO0 without shelly addon, and in this configuration setting GPIO0 to AM2301 seems to work correctly.
In the coming days I will carry out more precise tests.
Thanks and best regards.

**Comment by chivas03** [comment]
Hello,

can you maybe share your configuration of module parameters?
I have similar HW configuration: Shelly1 V3 + Add on. I can connect a DS18B20 and I receive OK values. I want to use as well DHT22 / AM302 but I am not able to get it done. I tried different module parameters but I don't get any values to the WebUI.

Regards,
Marco

**Comment by Massimo65** [comment]
Good morning,
The configuration is the standard shelly 1v3(46), I connected the sensor directly to the debug port +3.3v, gnd and gpio0. in the configuration I then set AM2301 to GPIO0. It works, I have a problem, sometimes after 2 hours, sometimes after 2 days the sensor disconnects and the only way to restore operation is to turn the shelly 1v3 off and on. Can you help me?

Thank you

**Reply by barbudor** [reply]
Similar problems have been already addressed multiple times
You need another GPIO to provide the power to the AM2301
Configure it as a Relay
When the AM stops responding power it off, wait then on again.
Using Pulsetime and PowerOnState 5 can help make it simpler.

**Reply by chivas03** [reply]
> Good morning, The configuration is the standard shelly 1v3(46), I connected the sensor directly to the debug port +3.3v, gnd and gpio0. in the configuration I then set AM2301 to GPIO0. It works, I have a problem, sometimes after 2 hours, sometimes after 2 days the sensor disconnects and the only way to restore operation is to turn the shelly 1v3 off and on. Can you help me?
> 
> Thank you

Hey,

thank you for your feedback. I want to avoid connecting the sensor directly to the board without the shelly add-on, as I use 230VAC for the power supply.  As I understood the communication between shelly and add-on is based on two-wire communication.
